source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.h

```diff
@@ -6,23 +6,39 @@
 #include "absl/base/thread_annotations.h"
 #include "absl/container/flat_hash_map.h"
 #include "absl/synchronization/mutex.h"
+#include "simple_lru_cache/simple_lru_cache_inl.h"
 
 // included to make code_format happy
 #include "envoy/extensions/cache/simple_http_cache/v3/config.pb.h"
 
+using ::google::simple_lru_cache::SimpleLRUCache;
+
+// The default maximum size of the cache is 100MiB
+constexpr uint64_t kSimpleHttpCacheDefaultSize = 100 * 1024 * 1024;
+
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace Cache {
 
 // Example cache backend that never evicts. Not suitable for production use.
 class SimpleHttpCache : public HttpCache {
+
 private:
   struct Entry {
     Http::ResponseHeaderMapPtr response_headers_;
     ResponseMetadata metadata_;
     std::string body_;
     Http::ResponseTrailerMapPtr trailers_;
+
+    // @return uint64_t calculates an approximated size (in bytes) by only
+    // measuring the bytes in the {header, body, trailers}. This does not account
+    // for data structures.
+    uint64_t byteSize() {
+      uint64_t headers_size = response_headers_ != nullptr ? response_headers_->byteSize() : 0;
+      uint64_t trailers_size = trailers_ != nullptr ? trailers_->byteSize() : 0;
+      return headers_size + sizeof(metadata_) + body_.size() + trailers_size;
```

Comment:
can do but I think the comment is still true regardless since even if we do `sizeof(*this)` we wouldn't be accounting for all the structures (response header / trailer are pointers here)